src/RealtimeEffectPanel.cpp

```diff
@@ -926,6 +954,35 @@ RealtimeEffectPanel::RealtimeEffectPanel(wxWindow* parent, wxWindowID id, const
    mEffectList = effectList;
 
    SetSizerAndFit(vSizer.release());
+
+   mTrackListChanged = TrackList::Get(mProject).Subscribe([this](const TrackListEvent& evt) {
+         auto track = evt.mpTrack.lock();
+         auto waveTrack = dynamic_cast<WaveTrack*>(track.get());
+
+         if (waveTrack == nullptr)
+            return;
+         
+         switch (evt.mType)
+         {
+         case TrackListEvent::TRACK_DATA_CHANGE:
+            if (mCurrentTrack.lock() == track)
+               mTrackTitle->SetLabel(track->GetName());
+            UpdateRealtimeEffectUIData(*waveTrack);
+            break;
+         case TrackListEvent::DELETION:
+            if (evt.mExtra == 0)
+               HideRealtimeUIForTrack(*waveTrack);
+            break;
+         case TrackListEvent::ADDITION:
+            // Addition can be fired as a part of "replace" event.
+            // Calling UpdateRealtimeEffectUIData is mostly no-op,
+            // it will just create a new State and Access for it.
+            UpdateRealtimeEffectUIData(*waveTrack);
```

Comment:
Aha, here it is then.  You are relying on the DELETION and ADDITION events both arriving, which is part of the contract in Track.h.  And you don't rely on the sequence of their arrival.